compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -339,31 +339,24 @@ void addVMVXDefaultPassPipeline(OpPassManager &passManager) {
   addBufferizePasses(nestedModulePM);
 }
 
-void addDoubleTilingExpertPassPipeline(OpPassManager &passManager,
-                                       bool enablePeeling, bool lowerToAVX2) {
+void addMultiTilingExpertPassPipeline(OpPassManager &passManager,
+                                      int64_t numLevel, bool enablePeeling,
+                                      bool lowerToAVX2) {
   addTileAndDistributePasses(passManager);
 
   OpPassManager &nestedModulePM = passManager.nest<ModuleOp>();
-  // Run LinalgFusePass firstly in case that we have fill + matmul + generic
-  // ops. At this stage, we do not apply vectorization. The reduction dim won't
-  // get tiled if the case is matmul + generic op. In this case, we have to tile
-  // along reduction dim again, which needs them to be Linalg ops form.
   {
     LinalgFusePassOptions options;
-    options.tilingLevel =
-        static_cast<int64_t>(StrategyTilingLevel::ParallelTiles);
-    nestedModulePM.addNestedPass<func::FuncOp>(createLinalgFusePass(options));
-    nestedModulePM.addNestedPass<func::FuncOp>(createCanonicalizerPass());
-    nestedModulePM.addNestedPass<func::FuncOp>(createCSEPass());
+    for (int64_t i = 1; i < numLevel; ++i) {
+      options.tilingLevel = i;
+      nestedModulePM.addNestedPass<func::FuncOp>(createLinalgFusePass(options));
+    }
   }
 
-  // Add the sandbox single tiling expert to tile and vectorize.
   {
     LinalgSingleTilingExpertPassOptions options;
     options.peel = enablePeeling;
     options.vectorize = true;
-    options.tilingLevel =
-        static_cast<int64_t>(StrategyTilingLevel::ReductionTiles);
     nestedModulePM.addNestedPass<func::FuncOp>(
```

Comment:
All the ops will be vectorized in this case. Because the anchorOpName is empty. In this case, the CodegenStrategy will look into all the Linalg ops and try to vectorize them.

(Actually, I'm surprised that the old code works... CodegenStrategy is a filter-based transform manager, each transform is applied to the ops that was changed/matched from the previous transform. I thought other ops (like linalg.fill, etc) do not have reduction dims. In this case, tiling won't put a filter/marker to the op, and vectorize should just skip them. Anyway, separating them into different CodegenStrategy passes look good to me. We aim to vectorize all the operations.)